nixos/modules/services/misc/nixos-needsreboot.nix

```diff
@@ -0,0 +1,24 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.nixos-needsreboot;
+in
+{
+  options = {
+    services.nixos-needsreboot = {
+      enable = lib.mkEnableOption "nixos-needsreboot, which determines if you need to reboot your NixOS machine";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.nixos-needsreboot = {
+      description = "Determine if you need to reboot your NixOS machine";
+      requires = [ "nixos-upgrade.service" ];
```

Comment:
So we should probably add an assertion that `config.system.autoUpgrade.enable == true`.

Reply:
Interesting, in which case, should that assertion be under `options`?

```diff
-      enable = lib.mkEnableOption "nixos-needsreboot, which determines if you need to reboot your NixOS machine";
+      enable = config.system.autoUpgrade.enable;
```

Reply:
It's definitely still needed. Not everyone enabling auto upgrades wants to use nixos-needsreboot.

Reply:
There is still no assertion.

Reply:
Oh, assertion. I was sleepy and misread+mistook it for depending on `config.system.autoUpgrade.enable`. Sorry, my bad lol

Reply:
So this module absolutely requires system.autoUpgrade? If not we could just make this conditional and I first thought that this would also work without auto upgrades.

Reply:
You can run the standalone binary from the shell if you want to manually trigger it, but to trigger the module, I don't see anything other than `system.autoUpgrade`.